tests/azure/availset_test.go

```diff
@@ -69,11 +69,11 @@ func TestAvailSetDeletion(t *testing.T) {
 
 	defer resources.Delete(t, resgroup)
 
-	availSets.Check(t, availset, resgroup)
+	availSets.AssertExists(t, availset, resgroup)
 
 	err = shell.Exec("TestAvailSetCreation", `
```

Comment:
I have a feeling that the err returned by shell.Exec will always be used to call t.Fatal... perhaps the shell.Exec could take t * testing.T as a parameter and explode if something goes wrong.